src/sentry/static/sentry/app/views/integrationOrganizationLink.tsx

```diff
@@ -0,0 +1,282 @@
+import {RouteComponentProps} from 'react-router/lib/Router';
+import React from 'react';
+import styled from '@emotion/styled';
+import {urlEncode} from '@sentry/utils';
+import {components} from 'react-select';
+
+import {Organization, IntegrationProvider} from 'app/types';
+import {addErrorMessage} from 'app/actionCreators/indicator';
+import {t, tct} from 'app/locale';
+import {
+  trackIntegrationEvent,
+  getIntegrationFeatureGate,
+} from 'app/utils/integrationUtil';
+import Alert from 'app/components/alert';
+import AsyncView from 'app/views/asyncView';
+import Button from 'app/components/button';
+import Field from 'app/views/settings/components/forms/field';
+import NarrowLayout from 'app/components/narrowLayout';
+import SelectControl from 'app/components/forms/selectControl';
+import IdBadge from 'app/components/idBadge';
+import LoadingIndicator from 'app/components/loadingIndicator';
+
+type Props = RouteComponentProps<{integrationSlug: string}, {}>;
+
+type State = AsyncView['state'] & {
+  selectedOrg?: string;
+  organization?: Organization;
+  provider?: IntegrationProvider;
+};
+
+//TODO: make generic props so we can use with Github external installation by extending class
```

Comment:
👎 to extending a react component. I would highly prefer composition over inheritance, so if anything I would say the TODO would be to split this into a reusable component.

Reply:
@EvanPurkhiser good call out, I'll change it